{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Publications markdown generator for academicpages\n",
    "\n",
    "Takes a TSV of publications with metadata and converts them for use with [academicpages.github.io](academicpages.github.io). This is an interactive Jupyter notebook ([see more info here](http://jupyter-notebook-beginner-guide.readthedocs.io/en/latest/what_is_jupyter.html)). The core python code is also in `publications.py`. Run either from the `markdown_generator` folder after replacing `publications.tsv` with one containing your data.\n",
    "\n",
    "TODO: Make this work with BibTex and other databases of citations, rather than Stuart's non-standard TSV format and citation style.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data format\n",
    "\n",
    "The TSV needs to have the following columns: pub_date, title, venue, excerpt, citation, site_url, and paper_url, with a header at the top. \n",
    "\n",
    "- `excerpt` and `paper_url` can be blank, but the others must have values. \n",
    "- `pub_date` must be formatted as YYYY-MM-DD.\n",
    "- `url_slug` will be the descriptive part of the .md file and the permalink URL for the page about the paper. The .md file will be `YYYY-MM-DD-[url_slug].md` and the permalink will be `https://[yourdomain]/publications/YYYY-MM-DD-[url_slug]`\n",
    "\n",
    "This is how the raw file looks (it doesn't look pretty, use a spreadsheet or other program to edit and create)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pub_date\ttitle\tvenue\texcerpt\tcitation\turl_slug\tpaper_url\r\n",
      "2019-01-04\tSignificant shared heritability underlies suicide attempt and clinically predicted probability of attempting suicide\tNature Molecular psychiatry\t\tDouglas Ruderfer,  Colin Walsh,  Matthew Aguirre,  Yosuke Tanigawa,  Jessica Ribeiro,  Joseph Franklin,  Manuel Rivas, \"Significant shared heritability underlies suicide attempt and clinically predicted probability of attempting suicide.\" Molecular psychiatry, 2019.\tSignificant-shared-heritability-underlies-suicide-attempt-and-clinically-predicted-probability-of-attempting-suicide\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC6609505/\r\n",
      "2018-03-19\tBayesian model comparison for rare variant association studies of multiple phenotypes\tbioRxiv\t\tChristopher DeBoever,  Matthew Aguirre,  Yosuke Tanigawa,  Chris Spencer,  Timothy Poterba,  Carlos Bustamante,  Mark Daly,  Matti Pirinen,  Manuel Rivas, \"Bayesian model comparison for rare variant association studies of multiple phenotypes.\" bioRxiv, 2018.\tBayesian-model-comparison-for-rare-variant-association-studies-of-multiple-phenotypes\thttps://doi.org/10.1101/257162 \r\n",
      "2017-10-12\tLandscape of X chromosome inactivation across human tissues\tNature\t\tTaru Tukiainen,  Alexandra-Chloe Villani,  Angela Yen,  Manuel Rivas,  Jamie Marshall,  Rahul Satija,  Matt Aguirre,  Laura Gauthier,  Mark Fleharty,  Andrew Kirby,  et. al., \"Landscape of X chromosome inactivation across human tissues.\" Nature, 2017.\tLandscape-of-X-chromosome-inactivation-across-human-tissues\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC5685192/\r\n",
      "2019-08-01\tPhenome-wide burden of copy-number variation in the UK biobank\tThe American Journal of Human Genetics\t\tMatthew Aguirre,  Manuel Rivas,  James Priest, \"Phenome-wide burden of copy-number variation in the UK biobank.\" The American Journal of Human Genetics, 2019.\tPhenome-wide-burden-of-copy-number-variation-in-the-UK-biobank\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC6699064/\r\n",
      "2018-12-05\tGlobal Biobank Engine: enabling genotype-phenotype browsing for biobank summary statistics\tBioinformatics\t\tGregory McInnes,  Yosuke Tanigawa,  Chris DeBoever,  Adam Lavertu,  Julia Olivieri,  Matthew Aguirre,  Manuel Rivas, \"Global Biobank Engine\tGlobal-Biobank-Engine:-enabling-genotype-phenotype-browsing-for-biobank-summary-statistics\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC6612820/\r\n",
      "2021-01-18\tGenetics of 35 blood and urine biomarkers in the UK Biobank\tNature Genetics\t\tNasa Sinnott-Armstrong,  Yosuke Tanigawa,  David Amar,  Nina Mars,  Christian Benner,  Matthew Aguirre,  Guhan Venkataraman,  Michael Wainberg,  Hanna Ollila,  Tuomo Kiiskinen,  et. al., \"Genetics of 35 blood and urine biomarkers in the UK Biobank.\" Nature Genetics, 2021.\tGenetics-of-35-blood-and-urine-biomarkers-in-the-UK-Biobank\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC7867639/\r\n",
      "2020-05-07\tAssessing digital phenotyping to enhance genetic studies of human diseases\tThe American Journal of Human Genetics\t\tChristopher DeBoever,  Yosuke Tanigawa,  Matthew Aguirre,  Greg McInnes,  Adam Lavertu,  Manuel Rivas, \"Assessing digital phenotyping to enhance genetic studies of human diseases.\" The American Journal of Human Genetics, 2020.\tAssessing-digital-phenotyping-to-enhance-genetic-studies-of-human-diseases\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC7212271/\r\n",
      "2019-09-06\tComponents of genetic associations across 2,138 phenotypes in the UK Biobank highlight adipocyte biology\tNature Communications\t\tYosuke Tanigawa,  Jiehan Li,  Johanne Justesen,  Heiko Horn,  Matthew Aguirre,  Christopher DeBoever,  Chris Chang,  Balasubramanian Narasimhan,  Kasper Lage,  Trevor Hastie,  et. al., \"Components of genetic associations across 2,138 phenotypes in the UK Biobank highlight adipocyte biology.\" Nature Communications, 2019.\tComponents-of-genetic-associations-across-2,138-phenotypes-in-the-UK-Biobank-highlight-adipocyte-biology\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC6731283/\r\n",
      "2021-02-08\tPolygenic risk modeling with latent trait-related genetic components\tEuropean Journal of Human Genetics\t\tMatthew Aguirre,  Yosuke Tanigawa,  Guhan Venkataraman,  Rob Tibshirani,  Trevor Hastie,  Manuel Rivas, \"Polygenic risk modeling with latent trait-related genetic components.\" European Journal of Human Genetics, 2021.\tPolygenic-risk-modeling-with-latent-trait-related-genetic-components\thttps://doi.org/10.1038/s41431-021-00813-0\r\n",
      "2020-10-23\tA fast and scalable framework for large-scale and ultrahigh-dimensional sparse regression with application to the UK Biobank\tPLoS genetics\t\tJunyang Qian,  Yosuke Tanigawa,  Wenfei Du,  Matthew Aguirre,  Chris Chang,  Robert Tibshirani,  Manuel Rivas,  Trevor Hastie, \"A fast and scalable framework for large-scale and ultrahigh-dimensional sparse regression with application to the UK Biobank.\" PLoS genetics, 2020.\tA-fast-and-scalable-framework-for-large-scale-and-ultrahigh-dimensional-sparse-regression-with-application-to-the-UK-Biobank\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC7641476/\r\n",
      "2020-11-04\tA deep learning classifier for local ancestry inference\tarXiv\t\tMatthew Aguirre,  Jan Sokol,  Guhan Venkataraman,  Alexander Ioannidis, \"A deep learning classifier for local ancestry inference.\" arXiv preprint arXiv\tA-deep-learning-classifier-for-local-ancestry-inference\thttps://arxiv.org/abs/2011.02081\r\n",
      "2020-11-23\tA phenome-wide association study of 26 mendelian genes reveals phenotypic expressivity of common and rare variants within the general population\tPLoS genetics\t\tCatherine Tcheandjieu,  Matthew Aguirre,  Stefan Gustafsson,  Priyanka Saha,  Praneetha Potiny,  Melissa Haendel,  Erik Ingelsson,  Manuel Rivas,  James Priest, \"A phenome-wide association study of 26 mendelian genes reveals phenotypic expressivity of common and rare variants within the general population.\" PLoS genetics, 2020.\tA-phenome-wide-association-study-of-26-mendelian-genes-reveals-phenotypic-expressivity-of-common-and-rare-variants-within-the-general-population\thttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC7735621/\r\n",
      "2020-11-25\tLearning epistatic polygenic phenotypes with Boolean interactions\tbioRxiv\t\tMerle Behr,  Karl Kumbier,  Aldo Cordova-Palomera,  Matthew Aguirre,  Euan Ashley,  Atul Butte,  Rima Arnaout,  James Brown,  James Preist,  Bin Yu, \"Learning epistatic polygenic phenotypes with Boolean interactions.\" bioRxiv, 2020.\tLearning-epistatic-polygenic-phenotypes-with-Boolean-interactions\thttps://doi.org/10.1101/2020.11.24.396846 \r\n",
      "2020-12-11\tIntegration of rare large-effect expression variants improves polygenic risk prediction\tmedRxiv\t\tCraig Smail,  Nicole Ferraro,  Matthew Durrant,  Abhiram Rao,  Matthew Aguirre,  Xin Li,  Michael Gloudemans,  Themistocles Assimes,  Charles Kooperberg,  Alexander Reiner,  et. al., \"Integration of rare large-effect expression variants improves polygenic risk prediction.\" medRxiv, 2020.\tIntegration-of-rare-large-effect-expression-variants-improves-polygenic-risk-prediction\thttps://doi.org/10.1101/2020.12.02.20242990 \r\n",
      "2021-04-22\tGenetic determinants of interventricular septal anatomy and the risk of ventricular septal defects and hypertrophic cardiomyopathy\tmedRxiv\t\tMengyao Yu,  Andrew Harper,  Matthew Aguirre,  Maurren Pittman,  Dulguun Amgalan,  Christopher Grace,  Anuj Goel,  Martin Farrall,  Ke Xiao,  Jesse Engreitz,  et. al., \"Genetic determinants of interventricular septal anatomy and the risk of ventricular septal defects and hypertrophic cardiomyopathy.\" medRxiv, 2021.\tGenetic-determinants-of-interventricular-septal-anatomy-and-the-risk-of-ventricular-septal-defects-and-hypertrophic-cardiomyopathy\thttps://doi.org/10.1101/2021.04.19.21255650 \r\n"
     ]
    }
   ],
   "source": [
    "!cat publications.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import pandas\n",
    "\n",
    "We are using the very handy pandas library for dataframes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import TSV\n",
    "\n",
    "Pandas makes this easy with the read_csv function. We are using a TSV, so we specify the separator as a tab, or `\\t`.\n",
    "\n",
    "I found it important to put this data in a tab-separated values format, because there are a lot of commas in this kind of data and comma-separated values can get messed up. However, you can modify the import statement, as pandas also has read_excel(), read_json(), and others."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pub_date</th>\n",
       "      <th>title</th>\n",
       "      <th>venue</th>\n",
       "      <th>excerpt</th>\n",
       "      <th>citation</th>\n",
       "      <th>url_slug</th>\n",
       "      <th>paper_url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2019-01-04</td>\n",
       "      <td>Significant shared heritability underlies suic...</td>\n",
       "      <td>Nature Molecular psychiatry</td>\n",
       "      <td></td>\n",
       "      <td>Douglas Ruderfer,  Colin Walsh,  Matthew Aguir...</td>\n",
       "      <td>Significant-shared-heritability-underlies-suic...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2018-03-19</td>\n",
       "      <td>Bayesian model comparison for rare variant ass...</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td></td>\n",
       "      <td>Christopher DeBoever,  Matthew Aguirre,  Yosuk...</td>\n",
       "      <td>Bayesian-model-comparison-for-rare-variant-ass...</td>\n",
       "      <td>https://doi.org/10.1101/257162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2017-10-12</td>\n",
       "      <td>Landscape of X chromosome inactivation across ...</td>\n",
       "      <td>Nature</td>\n",
       "      <td></td>\n",
       "      <td>Taru Tukiainen,  Alexandra-Chloe Villani,  Ang...</td>\n",
       "      <td>Landscape-of-X-chromosome-inactivation-across-...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2019-08-01</td>\n",
       "      <td>Phenome-wide burden of copy-number variation i...</td>\n",
       "      <td>The American Journal of Human Genetics</td>\n",
       "      <td></td>\n",
       "      <td>Matthew Aguirre,  Manuel Rivas,  James Priest,...</td>\n",
       "      <td>Phenome-wide-burden-of-copy-number-variation-i...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>Global Biobank Engine: enabling genotype-pheno...</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td></td>\n",
       "      <td>Gregory McInnes,  Yosuke Tanigawa,  Chris DeBo...</td>\n",
       "      <td>Global-Biobank-Engine:-enabling-genotype-pheno...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2021-01-18</td>\n",
       "      <td>Genetics of 35 blood and urine biomarkers in t...</td>\n",
       "      <td>Nature Genetics</td>\n",
       "      <td></td>\n",
       "      <td>Nasa Sinnott-Armstrong,  Yosuke Tanigawa,  Dav...</td>\n",
       "      <td>Genetics-of-35-blood-and-urine-biomarkers-in-t...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2020-05-07</td>\n",
       "      <td>Assessing digital phenotyping to enhance genet...</td>\n",
       "      <td>The American Journal of Human Genetics</td>\n",
       "      <td></td>\n",
       "      <td>Christopher DeBoever,  Yosuke Tanigawa,  Matth...</td>\n",
       "      <td>Assessing-digital-phenotyping-to-enhance-genet...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2019-09-06</td>\n",
       "      <td>Components of genetic associations across 2,13...</td>\n",
       "      <td>Nature Communications</td>\n",
       "      <td></td>\n",
       "      <td>Yosuke Tanigawa,  Jiehan Li,  Johanne Justesen...</td>\n",
       "      <td>Components-of-genetic-associations-across-2,13...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2021-02-08</td>\n",
       "      <td>Polygenic risk modeling with latent trait-rela...</td>\n",
       "      <td>European Journal of Human Genetics</td>\n",
       "      <td></td>\n",
       "      <td>Matthew Aguirre,  Yosuke Tanigawa,  Guhan Venk...</td>\n",
       "      <td>Polygenic-risk-modeling-with-latent-trait-rela...</td>\n",
       "      <td>https://doi.org/10.1038/s41431-021-00813-0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2020-10-23</td>\n",
       "      <td>A fast and scalable framework for large-scale ...</td>\n",
       "      <td>PLoS genetics</td>\n",
       "      <td></td>\n",
       "      <td>Junyang Qian,  Yosuke Tanigawa,  Wenfei Du,  M...</td>\n",
       "      <td>A-fast-and-scalable-framework-for-large-scale-...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2020-11-04</td>\n",
       "      <td>A deep learning classifier for local ancestry ...</td>\n",
       "      <td>arXiv</td>\n",
       "      <td></td>\n",
       "      <td>Matthew Aguirre,  Jan Sokol,  Guhan Venkataram...</td>\n",
       "      <td>A-deep-learning-classifier-for-local-ancestry-...</td>\n",
       "      <td>https://arxiv.org/abs/2011.02081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2020-11-23</td>\n",
       "      <td>A phenome-wide association study of 26 mendeli...</td>\n",
       "      <td>PLoS genetics</td>\n",
       "      <td></td>\n",
       "      <td>Catherine Tcheandjieu,  Matthew Aguirre,  Stef...</td>\n",
       "      <td>A-phenome-wide-association-study-of-26-mendeli...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2020-11-25</td>\n",
       "      <td>Learning epistatic polygenic phenotypes with B...</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td></td>\n",
       "      <td>Merle Behr,  Karl Kumbier,  Aldo Cordova-Palom...</td>\n",
       "      <td>Learning-epistatic-polygenic-phenotypes-with-B...</td>\n",
       "      <td>https://doi.org/10.1101/2020.11.24.396846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2020-12-11</td>\n",
       "      <td>Integration of rare large-effect expression va...</td>\n",
       "      <td>medRxiv</td>\n",
       "      <td></td>\n",
       "      <td>Craig Smail,  Nicole Ferraro,  Matthew Durrant...</td>\n",
       "      <td>Integration-of-rare-large-effect-expression-va...</td>\n",
       "      <td>https://doi.org/10.1101/2020.12.02.20242990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2021-04-22</td>\n",
       "      <td>Genetic determinants of interventricular septa...</td>\n",
       "      <td>medRxiv</td>\n",
       "      <td></td>\n",
       "      <td>Mengyao Yu,  Andrew Harper,  Matthew Aguirre, ...</td>\n",
       "      <td>Genetic-determinants-of-interventricular-septa...</td>\n",
       "      <td>https://doi.org/10.1101/2021.04.19.21255650</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      pub_date                                              title  \\\n",
       "0   2019-01-04  Significant shared heritability underlies suic...   \n",
       "1   2018-03-19  Bayesian model comparison for rare variant ass...   \n",
       "2   2017-10-12  Landscape of X chromosome inactivation across ...   \n",
       "3   2019-08-01  Phenome-wide burden of copy-number variation i...   \n",
       "4   2018-12-05  Global Biobank Engine: enabling genotype-pheno...   \n",
       "5   2021-01-18  Genetics of 35 blood and urine biomarkers in t...   \n",
       "6   2020-05-07  Assessing digital phenotyping to enhance genet...   \n",
       "7   2019-09-06  Components of genetic associations across 2,13...   \n",
       "8   2021-02-08  Polygenic risk modeling with latent trait-rela...   \n",
       "9   2020-10-23  A fast and scalable framework for large-scale ...   \n",
       "10  2020-11-04  A deep learning classifier for local ancestry ...   \n",
       "11  2020-11-23  A phenome-wide association study of 26 mendeli...   \n",
       "12  2020-11-25  Learning epistatic polygenic phenotypes with B...   \n",
       "13  2020-12-11  Integration of rare large-effect expression va...   \n",
       "14  2021-04-22  Genetic determinants of interventricular septa...   \n",
       "\n",
       "                                     venue excerpt  \\\n",
       "0              Nature Molecular psychiatry           \n",
       "1                                  bioRxiv           \n",
       "2                                   Nature           \n",
       "3   The American Journal of Human Genetics           \n",
       "4                           Bioinformatics           \n",
       "5                          Nature Genetics           \n",
       "6   The American Journal of Human Genetics           \n",
       "7                    Nature Communications           \n",
       "8       European Journal of Human Genetics           \n",
       "9                            PLoS genetics           \n",
       "10                                   arXiv           \n",
       "11                           PLoS genetics           \n",
       "12                                 bioRxiv           \n",
       "13                                 medRxiv           \n",
       "14                                 medRxiv           \n",
       "\n",
       "                                             citation  \\\n",
       "0   Douglas Ruderfer,  Colin Walsh,  Matthew Aguir...   \n",
       "1   Christopher DeBoever,  Matthew Aguirre,  Yosuk...   \n",
       "2   Taru Tukiainen,  Alexandra-Chloe Villani,  Ang...   \n",
       "3   Matthew Aguirre,  Manuel Rivas,  James Priest,...   \n",
       "4   Gregory McInnes,  Yosuke Tanigawa,  Chris DeBo...   \n",
       "5   Nasa Sinnott-Armstrong,  Yosuke Tanigawa,  Dav...   \n",
       "6   Christopher DeBoever,  Yosuke Tanigawa,  Matth...   \n",
       "7   Yosuke Tanigawa,  Jiehan Li,  Johanne Justesen...   \n",
       "8   Matthew Aguirre,  Yosuke Tanigawa,  Guhan Venk...   \n",
       "9   Junyang Qian,  Yosuke Tanigawa,  Wenfei Du,  M...   \n",
       "10  Matthew Aguirre,  Jan Sokol,  Guhan Venkataram...   \n",
       "11  Catherine Tcheandjieu,  Matthew Aguirre,  Stef...   \n",
       "12  Merle Behr,  Karl Kumbier,  Aldo Cordova-Palom...   \n",
       "13  Craig Smail,  Nicole Ferraro,  Matthew Durrant...   \n",
       "14  Mengyao Yu,  Andrew Harper,  Matthew Aguirre, ...   \n",
       "\n",
       "                                             url_slug  \\\n",
       "0   Significant-shared-heritability-underlies-suic...   \n",
       "1   Bayesian-model-comparison-for-rare-variant-ass...   \n",
       "2   Landscape-of-X-chromosome-inactivation-across-...   \n",
       "3   Phenome-wide-burden-of-copy-number-variation-i...   \n",
       "4   Global-Biobank-Engine:-enabling-genotype-pheno...   \n",
       "5   Genetics-of-35-blood-and-urine-biomarkers-in-t...   \n",
       "6   Assessing-digital-phenotyping-to-enhance-genet...   \n",
       "7   Components-of-genetic-associations-across-2,13...   \n",
       "8   Polygenic-risk-modeling-with-latent-trait-rela...   \n",
       "9   A-fast-and-scalable-framework-for-large-scale-...   \n",
       "10  A-deep-learning-classifier-for-local-ancestry-...   \n",
       "11  A-phenome-wide-association-study-of-26-mendeli...   \n",
       "12  Learning-epistatic-polygenic-phenotypes-with-B...   \n",
       "13  Integration-of-rare-large-effect-expression-va...   \n",
       "14  Genetic-determinants-of-interventricular-septa...   \n",
       "\n",
       "                                            paper_url  \n",
       "0   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...  \n",
       "1                     https://doi.org/10.1101/257162   \n",
       "2   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  \n",
       "3   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...  \n",
       "4   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...  \n",
       "5   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...  \n",
       "6   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...  \n",
       "7   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...  \n",
       "8          https://doi.org/10.1038/s41431-021-00813-0  \n",
       "9   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...  \n",
       "10                   https://arxiv.org/abs/2011.02081  \n",
       "11  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...  \n",
       "12         https://doi.org/10.1101/2020.11.24.396846   \n",
       "13       https://doi.org/10.1101/2020.12.02.20242990   \n",
       "14       https://doi.org/10.1101/2021.04.19.21255650   "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "publications = pd.read_csv(\"publications.tsv\", sep=\"\\t\", header=0).fillna(\"\")\n",
    "publications\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Escape special characters\n",
    "\n",
    "YAML is very picky about how it takes a valid string, so we are replacing single and double quotes (and ampersands) with their HTML encoded equivilents. This makes them look not so readable in raw format, but they are parsed and rendered nicely."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "html_escape_table = {\n",
    "    \"&\": \"&amp;\",\n",
    "    '\"': \"&quot;\",\n",
    "    \"'\": \"&apos;\"\n",
    "    }\n",
    "\n",
    "def html_escape(text):\n",
    "    \"\"\"Produce entities within text.\"\"\"\n",
    "    return \"\".join(html_escape_table.get(c,c) for c in text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating the markdown files\n",
    "\n",
    "This is where the heavy lifting is done. This loops through all the rows in the TSV dataframe, then starts to concatentate a big string (```md```) that contains the markdown for each type. It does the YAML metadata first, then does the description for the individual page."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "for row, item in publications.iterrows():\n",
    "    \n",
    "    md_filename = str(item.pub_date) + \"-\" + item.url_slug + \".md\"\n",
    "    html_filename = str(item.pub_date) + \"-\" + item.url_slug\n",
    "    year = item.pub_date[:4]\n",
    "    \n",
    "    ## YAML variables\n",
    "    \n",
    "    md = \"---\\ntitle: \\\"\"   + item.title + '\"\\n'\n",
    "    \n",
    "    md += \"\"\"collection: publications\"\"\"\n",
    "    \n",
    "    md += \"\"\"\\npermalink: /publication/\"\"\" + html_filename\n",
    "    \n",
    "    if len(str(item.excerpt)) > 5:\n",
    "        md += \"\\nexcerpt: '\" + html_escape(item.excerpt) + \"'\"\n",
    "    \n",
    "    md += \"\\ndate: \" + str(item.pub_date) \n",
    "    \n",
    "    md += \"\\nvenue: '\" + html_escape(item.venue) + \"'\"\n",
    "    \n",
    "    if len(str(item.paper_url)) > 5:\n",
    "        md += \"\\npaperurl: '\" + item.paper_url + \"'\"\n",
    "    \n",
    "    md += \"\\ncitation: '\" + html_escape(item.citation) + \"'\"\n",
    "    \n",
    "    md += \"\\n---\"\n",
    "    \n",
    "    ## Markdown description for individual page\n",
    "        \n",
    "    if len(str(item.excerpt)) > 5:\n",
    "        md += \"\\n\" + html_escape(item.excerpt) + \"\\n\"\n",
    "    \n",
    "    if len(str(item.paper_url)) > 5:\n",
    "        md += \"\\n[Read paper here](\" + item.paper_url + \")\\n\" \n",
    "        \n",
    "    md += \"\\nRecommended citation: \" + item.citation\n",
    "    \n",
    "    md_filename = os.path.basename(md_filename)\n",
    "       \n",
    "    with open(\"../_publications/\" + md_filename, 'w') as f:\n",
    "        f.write(md)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These files are in the publications directory, one directory below where we're working from."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2017-10-12-Landscape-of-X-chromosome-inactivation-across-human-tissues.md\r\n",
      "2018-03-19-Bayesian-model-comparison-for-rare-variant-association-studies-of-multiple-phenotypes.md\r\n",
      "2018-12-05-Global-Biobank-Engine-enabling-genotype-phenotype-browsing-for-biobank-summary-statistics.md\r\n",
      "2018-12-05-Global-Biobank-Engine:-enabling-genotype-phenotype-browsing-for-biobank-summary-statistics.md\r\n",
      "2019-01-04-Significant-shared-heritability-underlies-suicide-attempt-and-clinically-predicted-probability-of-attempting-suicide.md\r\n",
      "2019-08-01-Phenome-wide-burden-of-copy-number-variation-in-the-UK-biobank.md\r\n",
      "2019-09-06-Components-of-genetic-associations-across-2,138-phenotypes-in-the-UK-Biobank-highlight-adipocyte-biology.md\r\n",
      "2020-05-07-Assessing-digital-phenotyping-to-enhance-genetic-studies-of-human-diseases.md\r\n",
      "2020-10-23-A-fast-and-scalable-framework-for-large-scale-and-ultrahigh-dimensional-sparse-regression-with-application-to-the-UK-Biobank.md\r\n",
      "2020-11-04-A-deep-learning-classifier-for-local-ancestry-inference.md\r\n",
      "2020-11-23-A-phenome-wide-association-study-of-26-mendelian-genes-reveals-phenotypic-expressivity-of-common-and-rare-variants-within-the-general-population.md\r\n",
      "2020-11-25-Learning-epistatic-polygenic-phenotypes-with-Boolean-interactions.md\r\n",
      "2020-12-11-Integration-of-rare-large-effect-expression-variants-improves-polygenic-risk-prediction.md\r\n",
      "2021-01-18-Genetics-of-35-blood-and-urine-biomarkers-in-the-UK-Biobank.md\r\n",
      "2021-02-08-Polygenic-risk-modeling-with-latent-trait-related-genetic-components.md\r\n",
      "2021-04-22-Genetic-determinants-of-interventricular-septal-anatomy-and-the-risk-of-ventricular-septal-defects-and-hypertrophic-cardiomyopathy.md\r\n"
     ]
    }
   ],
   "source": [
    "!ls ../_publications/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---\r\n",
      "title: \"Polygenic risk modeling with latent trait-related genetic components\"\r\n",
      "collection: publications\r\n",
      "permalink: /publication/2021-02-08-Polygenic-risk-modeling-with-latent-trait-related-genetic-components\r\n",
      "date: 2021-02-08\r\n",
      "venue: 'European Journal of Human Genetics'\r\n",
      "paperurl: 'https://doi.org/10.1038/s41431-021-00813-0'\r\n",
      "citation: 'Matthew Aguirre,  Yosuke Tanigawa,  Guhan Venkataraman,  Rob Tibshirani,  Trevor Hastie,  Manuel Rivas, &quot;Polygenic risk modeling with latent trait-related genetic components.&quot; European Journal of Human Genetics, 2021.'\r\n",
      "---\r\n",
      "[Read paper here](https://doi.org/10.1038/s41431-021-00813-0)\r\n",
      "\r\n",
      "Recommended citation: Matthew Aguirre,  Yosuke Tanigawa,  Guhan Venkataraman,  Rob Tibshirani,  Trevor Hastie,  Manuel Rivas, \"Polygenic risk modeling with latent trait-related genetic components.\" European Journal of Human Genetics, 2021."
     ]
    }
   ],
   "source": [
    "!cat ../_publications/2021-02-08-Polygenic-risk-modeling-with-latent-trait-related-genetic-components.md"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
